pkg/reconciler/apiserversource/controller.go

```diff
@@ -67,7 +68,7 @@ func NewController(
 	deploymentInformer := deploymentinformer.Get(ctx)
 	apiServerSourceInformer := apiserversourceinformer.Get(ctx)
 	namespaceInformer := namespace.Get(ctx)
-	serviceaccountInformer := serviceaccountinformer.Get(ctx)
+	serviceaccountInformer := serviceaccountinformer.Get(ctx, sources.OIDCTokenRoleLabelSelector)
```

Comment:
just to not confuse other devs later (and maybe save some debugging time :D), I would rename `serviceaccountInformer` to something like `oidcServiceaccountInformer`. This should probably be done at the other places as well.